src/skins/vector/views/organisms/UserSettings.js

```diff
@@ -15,15 +15,144 @@ limitations under the License.
 
 var React = require('react');
 var sdk = require('matrix-react-sdk')
+var dis = require('matrix-react-sdk/lib/dispatcher')
 var MatrixClientPeg = require('matrix-react-sdk/lib/MatrixClientPeg');
+var q = require('q');
 
-var UserSettingsController = require('matrix-react-sdk/lib/controllers/organisms/UserSettings')
+var version = require('../../../../../package.json').version;
 
 var Modal = require('matrix-react-sdk/lib/Modal');
+var UserSettingsStore = require('matrix-react-sdk/lib/UserSettingsStore');
 
 module.exports = React.createClass({
     displayName: 'UserSettings',
-    mixins: [UserSettingsController],
+
+    Phases: {
+        Loading: "loading",
+        Saving: "saving",
+        Display: "display",
+    },
+
+    getInitialState: function() {
+        return {
+            avatarUrl: null,
+            displayName: null,
+            threePids: [],
+            clientVersion: version,
+            phase: this.Phases.Loading,
+        };
+    },
+
+    componentWillMount: function() {
+        var self = this;
+
+        var profilePromise = UserSettingsStore.loadProfileInfo();
+        var threepidPromise = UserSettingsStore.loadThreePids();
+
+        q.all([profilePromise, threepidPromise]).then(
```

Comment:
This will break subtly if the `onFulfilled` function throws. Please use `.done` instead of `.then` as it is the end of the promise chain.